From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Mon, 16 Feb 2026 17:09:13 +0100
Subject: [PATCH] Linux 6.12.73
References: bsc#1234429
Patch-mainline: 6.12.73
Git-commit: 5fb0303f6cb6a89bcfb19bd7a68cb793c86e78b2

Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Signed-off-by: Robert Frohl <rfrohl@suse.com>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 8af8b413c054..39a05470fbce 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 # SPDX-License-Identifier: GPL-2.0
 VERSION = 6
 PATCHLEVEL = 12
-SUBLEVEL = 72
+SUBLEVEL = 73
 EXTRAVERSION =
 NAME = Baby Opossum Posse
 
-- 
2.51.0

