From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Thu, 12 Feb 2026 13:14:57 +0100
Subject: [PATCH] Linux 6.12.71
References: bsc#1234429
Patch-mainline: 6.12.71
Git-commit: ae591174b1f2e6b81ffe182fb621bba910bfb44e

Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Signed-off-by: Robert Frohl <rfrohl@suse.com>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 3f2328829d28..443fa095f66f 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 # SPDX-License-Identifier: GPL-2.0
 VERSION = 6
 PATCHLEVEL = 12
-SUBLEVEL = 70
+SUBLEVEL = 71
 EXTRAVERSION =
 NAME = Baby Opossum Posse
 
-- 
2.51.0

