From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Fri, 27 Feb 2026 16:05:57 -0500
Subject: [PATCH] Linux 6.19.5
References: bsc#1012628
Patch-mainline: 6.19.5
Git-commit: c89ce241c1909d2c2bdde88334c33f3000d364fb

Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index e9396657c546..f486050e0bee 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 # SPDX-License-Identifier: GPL-2.0
 VERSION = 6
 PATCHLEVEL = 19
-SUBLEVEL = 4
+SUBLEVEL = 5
 EXTRAVERSION =
 NAME = Baby Opossum Posse
 
-- 
2.53.0

