From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Mon, 16 Feb 2026 17:11:45 +0100
Subject: [PATCH] Linux 6.19.2
References: bsc#1012628
Patch-mainline: 6.19.2
Git-commit: aa95a019990986bee50ff2bcede0185035b2e1e0

Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 965228c1c2ff..f0eb659930b2 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 # SPDX-License-Identifier: GPL-2.0
 VERSION = 6
 PATCHLEVEL = 19
-SUBLEVEL = 1
+SUBLEVEL = 2
 EXTRAVERSION =
 NAME = Baby Opossum Posse
 
-- 
2.52.0

