From: Vasant Karasulli <vkarasulli@suse.de>
Subject: netlink: Trim skb to alloc size to avoid MSG_TRUNC
Git-commit: db65a3aaf29ecce2e34271d52e8d2336b97bd9fe
Patch-mainline: v4.3-rc7
References: bsc#1166098

Signed-off-by: Vasant Karasulli <vkarasulli@suse.de>

--- a/net/netlink/af_netlink.c
+++ b/net/netlink/af_netlink.c
@@ -1694,6 +1694,9 @@
 	if (!skb)
 		goto errout;

+	/* Trim skb to allocated size to prevent MSG_TRUNC error */
+	skb_reserve(skb, skb_tailroom(skb) - alloc_size);
+
 	len = cb->dump(skb, cb);

 	if (len > 0) {
