From 2dbe93d1e4d730e6e730310035906aa85fa5eee6 Mon Sep 17 00:00:00 2001
From: Anish Dabhane <anishdabhane71@gmail.com>
Date: Sun, 18 May 2025 19:31:03 +0530
Subject: Fix spelling error for 'parallel'
Git-commit: 2dbe93d1e4d730e6e730310035906aa85fa5eee6
Patch-mainline: v6.16-rc1
References: jsc#PED-13979 jsc#PED-14039 jsc#PED-14046 jsc#PED-14211 jsc#PED-14333 jsc#PED-14487 jsc#PED-14488 jsc#PED-14497 jsc#PED-14499 jsc#PED-15868

Signed-off-by: Anish Dabhane <anishdabhane71@gmail.com>
Signed-off-by: Jonathan Corbet <corbet@lwn.net>
Message-id: <20250518140107.21310-1-anishdabhane71@gmail.com>
Acked-by: Patrik Jakobsson <pjakobsson@suse.de>

---
 Documentation/gpu/rfc/i915_scheduler.rst | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Documentation/gpu/rfc/i915_scheduler.rst b/Documentation/gpu/rfc/i915_scheduler.rst
index c237ebc024cd..2974525f0ac5 100644
--- a/Documentation/gpu/rfc/i915_scheduler.rst
+++ b/Documentation/gpu/rfc/i915_scheduler.rst
@@ -26,7 +26,7 @@ i915 with the DRM scheduler is:
 	  which configures a slot with N contexts
 	* After I915_CONTEXT_ENGINES_EXT_PARALLEL a user can submit N batches to
 	  a slot in a single execbuf IOCTL and the batches run on the GPU in
-	  paralllel
+	  parallel
 	* Initially only for GuC submission but execlists can be supported if
 	  needed
 * Convert the i915 to use the DRM scheduler
-- 
2.52.0

