Subject: hyperv_fb: disable rmmod
From: <ohering@suse.de>
Patch-mainline: Never, too many upstream changes
References: bsc#1241145, CVE-2025-21976

---
 drivers/video/fbdev/hyperv_fb.c |    1 -
 1 file changed, 1 deletion(-)

--- a/drivers/video/fbdev/hyperv_fb.c
+++ b/drivers/video/fbdev/hyperv_fb.c
@@ -1367,7 +1367,6 @@
 }
 
 module_init(hvfb_drv_init);
-module_exit(hvfb_drv_exit);
 
 MODULE_LICENSE("GPL");
 MODULE_DESCRIPTION("Microsoft Hyper-V Synthetic Video Frame Buffer Driver");
