From: Martin Wilck <mwilck@suse.com>
Subject: KABI: fix "Revert-dm-fix-a-race-condition-in-retrieve_deps.patch"
References: git-fixes

Signed-off-by: Martin Wilck <mwilck@suse.com>
Reviewed-by: Jan Kara <jack@suse.cz>
---
 drivers/md/dm-core.h |    2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/md/dm-core.h
+++ b/drivers/md/dm-core.h
@@ -223,6 +223,8 @@ struct dm_table {
 
 	/* a list of devices used by this table */
 	struct list_head devices;
+	/* KABI: unused, this member has been removed in upstream 4550a71b179b */
+	struct rw_semaphore devices_lock;
 
 	/* events get handed up using this callback */
 	void (*event_fn)(void *data);
